File: Backend/egd-user-with-quarkus/src/main/java/at/htl/model/Car.java
package at.htl.model;

import lombok.*;

import javax.persistence.*;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@ToString
@EqualsAndHashCode



@Entity
@Table(name = "car")
public class Car {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String licensePlate;

    private Double consumption;

    private Double longitude;
    private Double latitude;

    private String uuid;

    @ManyToOne
    @JoinColumn(name = "users_id")
    private Users currentUser;

   /* @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;
*/

}

File: AndroidProjekte/EGD/app/src/main/java/com/example/egd/data/entities/Car.kt
package com.example.egd.data.entities

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

data class Car(
    @SerializedName("id")
    @Expose
    var id:Long?,
    @SerializedName("name")
    @Expose
    var name:String,
    @SerializedName("consumption")
    @Expose
    var consumption:Double,
    @SerializedName("latitude")
    @Expose
    var latitude:Double,
    @SerializedName("longitude")
    @Expose
    var longitude:Double,
    @SerializedName("uuid")
    @Expose
    var uuid:String?,
    @SerializedName("licensePlate")
    @Expose
    var licensePlate: String,
    @SerializedName("currentUser")
    @Expose
    var currentUser: User?,
    var isAdmin: Boolean?,
    )
